File: csharp+code+xunit/CarPlant/Wheel.cs
namespace CarPlant {

	public class Wheel {

		private string name;

		public Wheel(string name) {
			this.name = name;
		}

		public void turnLeft(int degrees) {
			CarPlant.LOG += name + " turning left " + degrees + " degrees\n";
		}

		public void turnRight(int degrees) {
			CarPlant.LOG += name + " turning right " + degrees + " degrees\n";
		}

		public void rotate(int kph) {
			CarPlant.LOG += name + " accelerating " + kph + " kph\n";
		}

	}
}

File: java/src/com/xpsurgery/carplant/Wheel.java
package com.xpsurgery.carplant;

import java.io.IOException;
import java.io.Writer;

class Wheel {

	private String name;
  private Writer listener;

	Wheel(String name, Writer listener) {
		this.name = name;
    this.listener = listener;
	}

	void turnLeft(int degrees) {
		report(name + " turning left " + degrees + " degrees\n");
	}

	void turnRight(int degrees) {
		report(name + " turning right " + degrees + " degrees\n");
	}

	void rotate(int kph) {
		report(name + " accelerating " + kph + " kph\n");
	}

	private void report(String event) {
		try {
      listener.write(event);
    } catch (IOException e) { }
	}

}

